services/wallet/common/helpers.go

```diff
@@ -0,0 +1,32 @@
+package common
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/status-im/status-go/contracts/ierc20"
+)
+
+func PackApprovalInputData(amountIn *big.Int, approvalContractAddress *common.Address) ([]byte, error) {
```

Comment:
It might be silly since this function is straightforward, but I normally like writing a test for these kinds of things using real blockchain data. Basically get data for some approval from etherscan   and make sure the same input gives you the same output.

Reply:
@dlipicar test added, thanks.